src/modules/editor/sagas.ts

```diff
@@ -499,24 +497,27 @@ function createWearablePreviewChannel(controller: IPreviewController) {
 
 function* handleSetWearablePreviewController() {
   const controller: IPreviewController = yield select(getWearablePreviewController)
-  const emotesChannel = createWearablePreviewChannel(controller)
 
-  while (true) {
-    try {
-      const event: string = yield take(emotesChannel)
-      switch (event) {
-        case PreviewEmoteEventType.ANIMATION_PLAY:
-          yield put(setEmotePlaying(true))
-          break
-        case PreviewEmoteEventType.ANIMATION_PAUSE:
-          yield put(setEmotePlaying(false))
-          break
-        case PreviewEmoteEventType.ANIMATION_END:
-          yield put(setEmotePlaying(false))
-          break
+  if (controller) {
```

Comment:
This is just a question about the line 499, can't the controller be get from the action itself instead of the redux-state?

Reply:
yep, updating!